### DISCUSSION TITLE
Only calculating Anomalies till previous day for hourly granularity

### DISCUSSION THREAD

**Discussion by ArunMu21** [cuebook/CueObserve#201] [Q&A]
I am running the cueobserve for a dataset with hourly granularity but it does not seems to work on the current day data. Like If I am running the anomaly_definition today and giving the data till current hour, it is only calculating the anomalies till yesterday data.

Here is data I am Providing:
<img width="409" alt="Screenshot 2021-12-06 at 1 41 14 PM" src="https://user-images.githubusercontent.com/87470763/144810690-daeb22b1-5617-4eb1-9863-6e5e6446ce0d.png">

Here Is my query:
SELECT
TIMESTAMP_TRUNC(created_on, HOUR) as created_on, 
status, count(*) as count
from x.y 
where status in('SENT_TO_PROVIDER', 'DELIVERED') 
group by 1,2
order by 1 desc

Here Is the Anomaly Graph:

<img width="960" alt="Screenshot 2021-12-06 at 1 45 06 PM" src="https://user-images.githubusercontent.com/87470763/144811027-001122ca-d515-4ac7-a5bf-9c8c39403cde.png">

In this graph anomalies is calculated only till last day.. for the current day it is creating the dashed-line graph.

In my use case I wanted to check for the anomalies for current day data.

Is it something expected behaviour of Cueobserve or Am I making any mistake in query?

**Comment by sachinkbansal** [comment]
CueObserve assumes the last data point is partial and ignores it. In case of hourly anomalies, it should ignore the last hour data, not the last day's data. This is a known issue fixed by https://github.com/cuebook/CueObserve/issues/179.

Which version are you using?

**Reply by ArunMu21** [reply]
Hello @sachinkbansal , now I installed the latest version version of Cueobserve. Now while I am trying to execute anomaly definition, it is throwing below error:



[2021-12-10 10:31:18,068: ERROR/MainProcess] Connection to Redis lost: Retry (19/20) in 1.00 second.


[2021-12-10 10:31:19,073: CRITICAL/MainProcess]

Retry limit exceeded while trying to reconnect to the Celery redis result store backend. The Celery application must be restarted.



[2021-12-10 10:31:19,074: ERROR/MainProcess] Message Error: Couldn't apply scheduled task 1: Error 99 connecting to localhost:6379. Cannot assign requested address.

[' File "/opt/venv/bin/celery", line 8, in <module>\n sys.exit(main())\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/__main__.py", line 15, in main\n sys.exit(_main())\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/bin/celery.py", line 213, in main\n return celery(auto_envvar_prefix="CELERY")\n', ' File "/opt/venv/lib/python3.9/site-packages/click/core.py", line 829, in __call__\n return self.main(*args, **kwargs)\n', ' File "/opt/venv/lib/python3.9/site-packages/click/core.py", line 782, in main\n rv = self.invoke(ctx)\n', ' File "/opt/venv/lib/python3.9/site-packages/click/core.py", line 1259, in invoke\n return _process_result(sub_ctx.command.invoke(sub_ctx))\n', ' File "/opt/venv/lib/python3.9/site-packages/click/core.py", line 1066, in invoke\n return ctx.invoke(self.callback, **ctx.params)\n', ' File "/opt/venv/lib/python3.9/site-packages/click/core.py", line 610, in invoke\n return callback(*args, **kwargs)\n', ' File "/opt/venv/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func\n return f(get_current_context(), *args, **kwargs)\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/bin/base.py", line 132, in caller\n return f(ctx, *args, **kwargs)\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/bin/beat.py", line 72, in beat\n return beat().run()\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/apps/beat.py", line 77, in run\n self.start_scheduler()\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/apps/beat.py", line 105, in start_scheduler\n service.start()\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/beat.py", line 627, in start\n interval = self.scheduler.tick()\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/beat.py", line 339, in tick\n self.apply_entry(entry, producer=self.producer)\n', ' File "/opt/venv/lib/python3.9/site-packages/celery/beat.py", line 272, in apply_entry\n exc, traceback.format_stack(), exc_info=True)\n']

Traceback (most recent call last):

File "/opt/venv/lib/python3.9/site-packages/redis/connection.py", line 559, in connect

sock = self._connect()

File "/opt/venv/lib/python3.9/site-packages/redis/connection.py", line 615, in _connect

raise err

File "/opt/venv/lib/python3.9/site-packages/redis/connection.py", line 603, in _connect

sock.connect(socket_address)

OSError: [Errno 99] Cannot assign requested address


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

File "/opt/venv/lib/python3.9/site-packages/celery/backends/redis.py", line 117, in reconnect_on_error
